rsocket-core/src/main/java/io/rsocket/fragmentation/FragmentationDuplexConnection.java

```diff
@@ -16,115 +16,120 @@
 
 package io.rsocket.fragmentation;
 
+import static io.rsocket.util.AbstractionLeakingFrameUtils.toAbstractionLeakingFrame;
+import static io.rsocket.util.DisposableUtil.disposeQuietly;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.PooledByteBufAllocator;
 import io.rsocket.DuplexConnection;
 import io.rsocket.Frame;
-import io.rsocket.frame.FrameHeaderFlyweight;
-import org.jctools.maps.NonBlockingHashMapLong;
+import io.rsocket.util.AbstractionLeakingFrameUtils;
+import io.rsocket.util.NumberUtils;
+import java.util.Objects;
 import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
+import reactor.core.publisher.GroupedFlux;
 import reactor.core.publisher.Mono;
 
-/** Fragments and Re-assembles frames. MTU is number of bytes per fragment. The default is 1024 */
-public class FragmentationDuplexConnection implements DuplexConnection {
+/**
+ * A {@link DuplexConnection} implementation that fragments and reassembles {@link Frame}s.
+ *
+ * @see <a
+ *     href="https://github.com/rsocket/rsocket/blob/master/Protocol.md#fragmentation-and-reassembly">Fragmentation
+ *     and Reassembly</a>
+ */
+public final class FragmentationDuplexConnection implements DuplexConnection {
+
+  private final ByteBufAllocator byteBufAllocator;
+
+  private final DuplexConnection delegate;
 
-  private final DuplexConnection source;
-  private final NonBlockingHashMapLong<FrameReassembler> frameReassemblers =
-      new NonBlockingHashMapLong<>();
   private final FrameFragmenter frameFragmenter;
 
-  public FragmentationDuplexConnection(DuplexConnection source, int mtu) {
-    this.source = source;
-    this.frameFragmenter = new FrameFragmenter(mtu);
+  /**
+   * Creates a new instance.
+   *
+   * @param delegate the {@link DuplexConnection} to decorate
+   * @param maxFragmentSize the maximum fragment size
+   * @throws NullPointerException if {@code byteBufAllocator} or {@code delegate} are {@code null}
```

Comment:
nit: byteBufAllocator not provided

Reply:
No problems with nits.  I’m super-meticulous about my code, and especially my documentation, so keep up with this level of review.